### PR TITLE
[Ready for review] Added minesweeper to arcade machines

### PR DESCRIPTION
**Pull request by Jujumatic**
ready plox review
also fuck tgui so i didn't use it for this thing

you get to play hecking minesweeper dude!!! how cool is that!?!?

things to fix:
- ~~get the square id generation actually working properly, cause it isn't working~~
- ~~href exploits and ways to hack the game into winning~~

things to add:
- ~~add emag effects e.g. hitting a mine causes a random explosion nearby or... somewhere~~
- ~~add a flag button to mark mines~~ (done and... as convenient to use as i can possibly make it!)
- ~~make the window actually presentable and look ok~~
- ~~add graphics for squares and mines~~
- ~~add sounds~~
- ~~syndicate bombs defusing is replaced by minesweeper OR minesweeper can be selected as an alternative to cutting wire rng bullshit~~ (this should be done in another pr so it can be reviewed)
- ~~create a limit to how many rows and columns you can make in custom... as hilarious as it is, grid generation lags the server to shit when over 50 rows and columns are involved~~ column limit of 50 and row limit of 30, mine limit is set to a percentage of the rows and columns made. does not lag!

shoutouts to this channel https://www.youtube.com/channel/UCA8luKdJ7MAFj4-LB3gFjXQ for making incredibly good arcade sounds

Changelog
:cl:
add: Added minesweeper to arcade games!
/:cl:

### CODE REVIEW

**Comment by Jujumatic** [comment]
also minesweeper is probably a copyrighted name so i'll change it to a different name later i guess

**Comment by k-k-karma** [comment]
it's not

**Comment by Jujumatic** [comment]
Great!

**Comment by Lxtoofonea** [comment]
+1

**Comment by Carbonhell** [comment]
+1
Reviewing this code will not be fun

**Comment by TheMeatGod** [comment]
+1, neat little thing to have

**Comment by earthmoonjune** [comment]
+1

**Comment by Manly-Dwarf** [comment]
this is already approved, do I have to vote here? does it matter?
smelly @HideAndSeekLOGIC

**Comment by Maroonenahmoron** [comment]
+1, a game that's worth playing inside of this game

**Comment by Jujumatic** [comment]
I believe this is ready for codechecking, enjoy!

**Comment by ghost** [comment]
@Jujumatic the travis failed. Fuck

**Comment by ghost** [comment]
fuck, the rocket pr

**Comment by ghost** [comment]
oh god, maybe not today juju

**Comment by Jujumatic** [comment]
Definitely not today because the rocket pr and I'm seeing just a few more things I can optimise a tad anyway ;)

**Comment by Jujumatic** [comment]
ready for review for good this time. big testing done, i think i got everything. except maybe href exploits idk

**Comment by Jujumatic** [comment]
checks passed... get to work boys......

File: hippiestation/code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -49,3 +49,404 @@
...
+			web += "</tr>"
+		web += "</table>"
+		web += "</tbody>"
+	web += "<br>"
+
+	if(game_status == MINESWEEPER_GAME_LOST)
+		web += "<font size='6'>You have lost!<br><font size='3'>Try again?<br><b><a href='byond://?src=[REF(src)];Easy=1'><font color='#cc66ff'>Easy (9x9 board, 10 mines)</font></a><br><a href='byond://?src=[REF(src)];Intermediate=1'><font color='#cc66ff'>Intermediate (16x16 board, 40 mines)</font></a><br><a href='byond://?src=[REF(src)];Hard=1'><font color='#cc66ff'>Hard (16x30 board, 99 mines)</font></a><br><a href='byond://?src=[REF(src)];Custom=1'><font color='#cc66ff'>Custom</font></a></b><br><a href='byond://?src=[REF(src)];same_board=1'><font color='#cc66ff'>Play on the same board</font></a><br><a href='byond://?src=[REF(src)];Main_Menu=1'><font color='#cc66ff'>Return to Main Menu</font></a></b><br>"
+
+	if(game_status == MINESWEEPER_GAME_WON)
+		if(rows < 10 && columns < 10)	//If less than easy difficulty
```
Comment (k-k-karma) [inline hippiestation/code/game/machinery/computer/arcade.dm]: this should be an or

Comment (Jujumatic) [reply hippiestation/code/game/machinery/computer/arcade.dm]: fixed

**Pull request merged**